oasis/cloud-migration-tool.md

```diff
@@ -0,0 +1,228 @@
+  --oasis.api-secret=$OASIS_API_SECRET \
+  --oasis.deployment-id=$OASIS_DEPLOYMENT_ID
+```
+
+You can also add the `--watch` option to start monitoring the status in real-time.
+
+### Stopping the migration process
+
+The `arangosync-migration stop` command stops the migration and terminates
+the migration agent process.
+
+If replication is running normally, the command waits until all shards are
+in sync. The local cluster is then switched into read-only mode.
+After all shards are in-sync and the migration stopped, the target deployment
+is switched into normal mode (read/write) while the source cluster stays in
+read-only mode. 
+
+```bash
+./arangosync-migration stop \
+  --oasis.api-key=$OASIS_API_KEY \
+  --oasis.api-secret=$OASIS_API_SECRET \
+  --oasis.deployment-id=$OASIS_DEPLOYMENT_ID
+```
+
+An optional `--abort` option is supported. If specified, the `stop` command 
+will not check anymore if both deployments are in-sync and stops all
+migration-related processes as soon as possible.
+
+### Switching the local cluster to read-only mode
+
+The `arangosync-migration set-server-mode` command allows switching [read-only mode](../http/administration-and-monitoring.html#update-whether-or-not-a-server-is-in-read-only-mode)
+for your local cluster on and off.
+
+In a read-only mode, all write operations are going to fail with an error code
+of `1004` (ERROR_READ_ONLY).
+Creating or dropping databases and collections are also going to fail with 
+error code `11` (ERROR_FORBIDDEN).
+
+```bash
+./arangosync-migration set-server-mode \
+  --source.cacert=tls-ca.crt \
+  --source.keyfile=client-auth.keyfile \
+  --source.endpoint=$COORDINATOR_ENDPOINT \
+  --source.jwtSecret=clusterSecret \
+  --source.server-mode=readonly
+```  
+The `--source.server-mode` option allows you to specify the desired server mode.
+Allowed values are `readonly` or `default`.
+
+## Cloud migration workflow for minimal downtime
+
+1. Download and start the `arangosync-migration` tool. The target deployment
+   is switched into read-only mode automatically.
+2. Wait until all shards are in sync. You can use the `status` or the `start`
+   command with the same parameters to track that.
+3. Optional: when all shards are in-sync, you can switch your applications
+   to use the Oasis deployment, but note that it stays into read-only mode
+   until the migration process is fully completed.
+4. Stop the migration using the `stop` subcommand. What happens:
+   - The source data cluster is switched into read-only mode.
+   - It waits until all shards are synchronized.
+   - The target deployment is switched into default read/write mode.
+
+   {% hint 'info' %}
+   After finishing the migration, the source data cluster will remain read-only. 
+   You can use the `set-server-mode` 
+   subcommand to switch it back to default, if needed.
+   In case something goes wrong during the migration, the `stop` command is not
+   switching the source data cluster into read-only mode. 
```

Comment:
How can I execute a 'dry-run', so a try to move everything into oasis without my source deployment does into RO mode. This can be good for an initial test phase. However I want to end-up with a non RO Oasis deployment as well, so I can execute test on that.

Reply:
@Robert-Stam - as already discussed before - 'dry-run' is not implemented for now, unfortunately. That is for next iterations. And target deployment of course will go into default read/write mode after stopping the sync.